core/controllers/acl_decorators.py

```
@@ -5133,6 +5133,71 @@ def test_can_access_certificate_dashboard(
+        self: _SelfBaseHandlerType, question_id: str, **kwargs: Any
+    ) -> _GenericHandlerFunctionReturnType:
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        from core.storage.certificate_assessment import gae_models
+
+        attempt_model = gae_models.CertificateAssessmentAttemptModel.query(
+            gae_models.CertificateAssessmentAttemptModel.learner_id
+            == self.user_id,
+            gae_models.CertificateAssessmentAttemptModel.is_submitted
+            == False,  # pylint: disable=singleton-comparison
```

Ruff (0.16.0)

[error] Avoid equality comparisons to `False`; use `not ...:` for false checks

Replace comparison

(E712)